### ISSUE TITLE
extending the max term?

### ISSUE THREAD

**Issue by philbertw4** [defiads/defiads#19]
Sorry for opening so many issues. While trying to debug my other issue #18, I came across this:
https://github.com/defiads/defiads/blob/f3f830a045d7bf32e40d67609a02cea9ea0e78d7/src/wallet.rs#L36 which indicates that the maximum term length that this reference implementation wallet would fund is approximately 1 month? This seems arbitrarily short. Is there a particular reason why it is necessary to have a max term length anything less than what the underlying bitcoin pay-to-contract protocol will support?

Or, perhaps, as it stands right now, this max term limit is just a temporary implementation detail that we can eventually remove.

**Comment by ZmnSCPxj** [comment]
A term length of e.g. 1000 years would be indistinguishable from burning the funds, as Bitcoin may not exist for that long (and even if it does, whoever owns it would have long ceased to remain the same entity by then).

**Comment by philbertw4** [comment]
Right, I tend to agree, and sorry to belabor this a bit. However, I think we should at least consider a max term length which is at least as long as (preferably longer than) fiat government bonds. There may not be demand for such long time frames initially, but I'm not sure that it makes sense to preclude such time frames. A max term length of 30, 50, 100, or even 200 years, while seemingly outlandish to us right now, may end up with some interesting use cases in not too distant future.

**Comment by philbertw4** [comment]
Circling back to this issue. Have you had a chance @ZmnSCPxj to think about it a bit more at all? If we restrict ads to 30 days, the project is of course still interesting, but, for me at least, distinctly less so than if we allow for longer terms.

**Comment by ZmnSCPxj** [comment]
I think no more than a year would be reasonable.  If you lock for more than that, it seems implausible to me that the economic conditions that made you put that ad up in the first place will remain the same.  With a 1-year limit, ad renewals can be as low as once a year, so should be acceptable in terms of blockchain bloat.
